log/logtest/recorder.go

```diff
@@ -141,22 +118,55 @@ func (r *Recorder) Result() []*ScopeRecords {
 	defer r.mu.Unlock()
 
 	ret := []*ScopeRecords{}
-	ret = append(ret, r.currentScopeRecord)
 	for _, l := range r.loggers {
-		ret = append(ret, l.Result()...)
+		ret = append(ret, l.scopeRecord)
 	}
 	return ret
 }
 
-// Reset clears the in-memory log records.
+// Reset clears the in-memory log records for all loggers.
 func (r *Recorder) Reset() {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	if r.currentScopeRecord != nil {
-		r.currentScopeRecord.Records = nil
-	}
 	for _, l := range r.loggers {
 		l.Reset()
 	}
 }
+
+type logger struct {
+	embedded.Logger
+
+	mu          sync.Mutex
+	scopeRecord *ScopeRecords
+
+	// enabledFn decides whether the recorder should enable logging of a record or not
+	enabledFn enabledFn
+}
+
+// Enabled indicates whether a specific record should be stored.
+func (l *logger) Enabled(ctx context.Context, record log.Record) bool {
+	if l.enabledFn == nil {
+		return defaultEnabledFunc(ctx, record)
+	}
+
+	return l.enabledFn(ctx, record)
+}
+
+// Emit stores the log record.
+func (l *logger) Emit(_ context.Context, record log.Record) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	l.scopeRecord.Records = append(l.scopeRecord.Records, record)
```

Comment:
Should this also check if `l.scopeRecord` is not `nil` before derefing?

Reply:
We can, but logger is a private struct which is always created through `Logger(`, and never have a nil scoperecord.
If it is indeed nil, we'd probably not want to silence it either (and this is a test package, so a panic wouldn't impact production code).

Reply:
Sounds good leaving it out then.

Should the nil check be removed from `Reset`?